backend/ai_firm/report_generation.py

```
@@ -705,6 +707,33 @@ def generate_full_report(self, symbol: str) -> Dict[str, Any]:
+
+            if loop.is_running():
+                # If running, we might need to use nest_asyncio or just fallback/mock if we can't await
+                # For this script context, usually it's fine.
+                # But to be safe, if we are in a running loop (e.g. uvicorn), we should ideally be async.
+                # Since this method is sync, we might skip async calls or use a separate thread.
+                # Fallback for simplicity in this context:
+                pass
+            else:
+                macro_liquidity = loop.run_until_complete(self.perplexity.get_macro_liquidity())
+                causality_data = loop.run_until_complete(self.perplexity.get_cross_asset_causality())
+        except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)